incubator/hnc/internal/validators/object.go

```diff
@@ -111,18 +112,31 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	// If the object wasn't and isn't inherited, we will check to see if the
 	// source can be created without causing any conflict.
 	if !oldInherited && !newInherited {
+		// check selector format
+		oldSelectorStr, _ := selector.GetSelectorAnnotation(oldInst)
+		newSelectorStr, _ := selector.GetSelectorAnnotation(inst)
+		sel := selector.GetSelectorFromString(log, newSelectorStr)
+		if oldSelectorStr != newSelectorStr && sel == nil {
+			return deny(metav1.StatusReasonInvalid, "Invalid \""+api.AnnotationSelector+"\" value: "+newSelectorStr+
+				", which can not be parsed as a labelSelector")
```

Comment:
Instead of `, which can not be parsed as a labelSelector`, I'd recommend ` is not a valid Kubernetes labelSelector`. If you Google "kubernetes labelselector" you immediately see the correct documentation as the first result, and this makes it clear that it's a K8s concept that the user needs to learn about, not an HNC concept.

Also, "can not be parsed as" suggests there might be an error with the parser, but "is not a valid labelselector" makes it clear that the user has put in a bad value.

The error returned by the parser would probably be very useful here.

Finally, it probably makes sense to change this to a format string which would be much easier to read:

```
msg := fmt.Sprintf("Invalid %s value: %q is not a valid Kubernetes labelSelector: %s", api.AnnotationSelector, newSelectorStr, err)
```
```

Reply:
done